画像読み込み関数の追加

hiroto1130 opened the pull request:
## 元issue

#3 

## 内容
- 画像読み込み関数の追加
  - Loadtexture関数の作成
  - TextureData構造体の作成
  - CUSTOMVERTEX構造体の作成
  - Class MapChipDataの作成

hiroto1130 committed:
画像読み込み関数の作成、追加(#3)

hiroto1130 committed:
global関数のクラス内関数化、クラスのカプセル化、MApChipData.h の追加(#3)

OkamotoKazuma committed:
競合の解決

taishi-matsumura commented:
> 修正してみました、どうでしょうか？

いいと思います 👍

taishi-matsumura commented on Thirteen/Texture.cpp:
```diff
@@ -0,0 +1,30 @@
+static TEXTUREDATA** TextureData[MAXSCENE] = { 0 };
+
+// �摜�ǂݍ��݊֐�
+void LoadTexture(const char* file_name, int SelectScene, int  SelectTexture)
```
なぜglobalな関数？

taishi-matsumura commented on Thirteen/Texture.h:
```diff
@@ -1 +1,53 @@
+		tv;	// ����
+} CUSTOMVERTEX, * P_CUSTOMVERTEX;
+
+class MapChipData
```
ファイル名とクラス名が別な理由は？

taishi-matsumura commented on Thirteen/Texture.h:
```diff
@@ -1,5 +1,5 @@
-#ifndef DRAW_H
-#define DRAW_H
+#ifndef TEXTURE_H
+#define TEXTURE_H
```
ファイル末尾のコメントも修正すべきですね。

taishi-matsumura commented on Thirteen/Texture.cpp:
```diff
@@ -0,0 +1,30 @@
+#include"SceneSwitch.h"
+
+MapChipData chipdata;
+static LPDIRECT3DDEVICE9 pDevice;
```
無名のnamespaceに入れるとかしましょう。

参考：https://ttsuki.github.io/styleguide/cppguide.ja.html#Unnamed_Namespaces_and_Static_Variables

hiroto1130 replied:
今回は使用しない方向で行くそうなので、勉強だけしておきます

hiroto1130 replied:
一通り修正しました

taishi-matsumura reviewed (changes requested):
たまたま見ちゃったので指摘しておきます。

taishi-matsumura commented on Thirteen/MapchipData.h:
```diff
@@ -0,0 +1,25 @@
+	const int Map_WidthNumber = 10;
+
+
+	float MapChipHeight;
```
publicなメンバ変数？

taishi-matsumura reviewed (changes requested):
カプセル化しないの？

taishi-matsumura commented on Thirteen/MapchipData.h:
```diff
@@ -0,0 +1,25 @@
+﻿#ifndef MAPCHIPDATA_H
+#define MAPCHIPDATA_H
```
全部大文字or小文字の場合は単語の区切りがわからなくなるのでsnakeにすることが多いです。

taishi-matsumura commented on Thirteen/MapchipData.h:
```diff
@@ -1,25 +1,28 @@
-﻿#ifndef MAPCHIPDATA_H
-#define MAPCHIPDATA_H
+﻿#ifndef MapChipData_H
+#define MapChipData_H
```
これはsnakeじゃないですね

taishi-matsumura commented on Thirteen/Texture.h:
```diff
@@ -25,22 +25,22 @@ class MapChipData
-	float Texture_Widht;
-	void InputTextureSize(int Height, int Widht)
+	float Texture_Width;
+	void InputTextureSize(int Height, int Width)
```
単純にメンバ変数を設定する関数では無いですが、それでも `InputXXXX()` よりは `SetXXX()` の方がマシだと思います。

hiroto1130 committed:
inputXXX()をSetXXX()に変更(#3)

- InputMapChipSize を SetMapChipSize に変更
- InputTextureSize を SetTextureSize に変更

taishi-matsumura commented:
例えば直近のcommitであれば、
```
InputXXX()をSetXXX()に変更

- InputMapChipSize を SetMapChipSize に変更
- InputTextureSize を SetTextureSize に変更
```
とかじゃないですか？

「titleとして用いられる1行目と、詳細として用いられる3行目以後が同じ概念レベルなのは変だな」という感覚を早く身に付けましょう。
日本語の文章で概念レベルを合わせられない人は関数も概念レベルがバラバラのものを混在させがちで、極めて読みづらいコードを書きます。迷惑です。
